### PR TITLE
Add lila github action

### PR DESCRIPTION
**Pull request by fitztrev**
This adds a CI workflow that runs an instance of lila within. It uses lila-db-seed to create users, api tokens, games, etc.

This could be used to test Berserk by having it make actual requests and then verifying the response. I included a sample script as a proof of concept.

The image is created from here: https://github.com/lichess-org/lila-docker/pull/62 It publishes the image to Github's container registry, which is where this github action then pulls it from.

### CODE REVIEW

**Comment by kraktus** [comment]
Very nice! This could help for https://github.com/lichess-org/berserk/issues/65 (checking in things break) with head, mainly.

The only caveat I see is the inability to run it locally(?).

**Comment by fitztrev** [comment]
> The only caveat I see is the inability to run it locally(?).

Correct. I'll see if I can improve the DX.

**Comment by fitztrev** [comment]
Updated it to work locally as well as with Github Actions.

The local workflow is designed to not conflict with another local instance of lila you may have running. The tests run in containers within their own bridge network to mimic the Github Actions workflow.

Each time the tests run, it's a clean instance of lila and a fresh lila-db-seed database. The tests can change the database and it will be reset on the next run.

To run locally:

```bash
./integration/local
```

File names and structure can all change. Just tried to get things working. There are now 2 types of Pytest files: the existing ones from `/tests` and now `/integration`. Added 2 sample test files.

**Comment by kraktus** [comment]
Hey, haven't forgotten about this, will check later this week.

File: integration/local [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    docker rm --force app > /dev/null 2>&1 || true
+    docker network rm lila-network > /dev/null 2>&1 || true
+}
+
+echo "Running integration tests"
+cleanup_containers
+
+docker network create lila-network
+docker run --name lila --network lila-network -d $IMAGE
+docker run --name app --network lila-network -v $(pwd):/app -w /app $IMAGE ./integration/run-tests.sh
```
Comment (kraktus) [inline integration/local:16]: Sorry for the delay! Finally looking at this, testing work. I don't understand why we need 2 containers?

Comment (kraktus) [reply integration/local:16]: Ah I see, one is running Lila, the other only the test, I was wondering since they both use the same image

**Comment by kraktus** [comment]
Thanks!

**Pull request merged**